server/controllers/finance/accounts/index.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * @module Accounts
+ *
+ * @description
+ * Implements CRUD operations on the Account entity.
+ *
+ * This module implements the following routes:
+ *  GET    /accounts
+ *  GET    /accounts/:id
+ *  GET    /accounts/:id/balance/
+ *  POST   /accounts
+ *  PUT    /accounts/:id
+ *
+ * @todo - move away from calling lookup() before action.  This is an
+ * unnecessary database request.
+ *
+ * @requires db
+ * @requires NotFound
+ * @requires accounts/types
+ */
+
+'use strict';
+
+const db = require('../../../lib/db');
+const NotFound = require('../../../lib/errors/NotFound');
+const types = require('./types');
+
+/**
+ * @method create
+ *
+ * @description
+ * Create a new account entity in the database.
+ *
+ * POST /accounts
+ */
+function create(req, res, next) {
+  let data = req.body;
+  let sql = 'INSERT INTO account SET ?';
+
+  delete data.id;
+
+  db.exec(sql, [data])
+    .then(function (result) {
+      res.status(201).json({id : result.insertId});
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+ * @method update
+ *
+ * @description
+ * Updates an account in the database.
+ *
+ * PUT /accounts/:id
+ */
+function update(req, res, next) {
+  let id = req.params.id;
+  let data = req.body;
+  let sql = 'UPDATE account SET ? WHERE id = ?';
+
+  delete data.id;
+
+  lookupAccount(id)
+    .then(function () {
+      return db.exec(sql, [data, id]);
+    })
+    .then(function() {
+      return lookupAccount(id);
+    })
+    .then(function (account) {
+      res.status(200).json(account);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+ * @method list
+ *
+ * @description
+ * Lists all accounts in the database
+ *
+ * GET /accounts/:id
+ */
+function list(req, res, next) {
+  let sql =
+    'SELECT a.id, a.number, a.label, a.locked FROM account AS a';
+
+  if (req.query.detailed === '1') {
+    sql = `
+      SELECT a.id, a.enterprise_id, a.locked, a.cc_id, a.pc_id, a.created,
+        a.classe, a.is_asset, a.reference_id, a.is_brut_link, a.is_charge,
+        a.number, a.label, a.parent, a.type_id, a.is_title, at.type
+      FROM account AS a JOIN account_type AS at ON a.type_id = at.id
+    `;
+  }
+
+  if (req.query.locked === '0') {
```

Comment:
Does the implementation of this flag make sense from an API standpoint? 

For example: 
```js
// returns only accounts that are not locked
'/accounts/:id?locked=0' 

// returns all accounts without considering locked property
'/accounts/:id?locked=1'
```

Either flags like this could implement both behaviours or it could be more of a binary flag 

For example:
```js
'/accounts/:id?activeAccounts=1'
```

What do you think?